styles-client/less-api.less

```diff
@@ -0,0 +1,3 @@
+.markdown-preview-plus() {}
+@import (optional, reference) '../../../styles.less';
```

Comment:
The problem with this approach is that it assumes a particular directory layout, which might not necessarily be like that. For instance, I have MPP symlinked into `~/.atom/`, so this `../../../` would in fact lead to the directory containing my work projects. Not good.

Besides, are you sure this will work as intended? `getClientStyle` will compile this stylesheet to CSS and then push the resulting CSS onto the preview. For one, this would lead to user stylesheet being pushed twice. For two, this just looks horribly convoluted.

Reply:
I thought about realtive path problem. It can be fixed via passing variable from js to less somehow (writing to file may be like [here](https://github.com/kiwi0fruit/misc/commit/70609c253629c09d51dc35076e69755d31d2d869)). Or we can test if 
```less
.markdown-preview-plus() {}
@import (optional, reference) '../../../styles.less';
@import (optional, reference) '~/.atom/styles.less';
.markdown-preview-plus();
``` 
would solve Unix use-cases (is can't be fixed on Windows this way, `%USERPROFILE%` doesn't work).

Anyway, with `(optional)` option in case of non-default package folder we simply get empty css file without popping errors.

And because of the `(reference)` option only `.markdown-preview-plus();` will be compiled - that is the output of the function defined in `styles.less` (but it doesn't affect css made from `styles.less` as it should not be called there). Other contents of the `styles.less` won't be in the resulting file. So there would not be any doubling of styles.

I've tested it and it works as intended. And as about how it looks from developer's perspective: it looks neat for me :) But who cares? What's important: it gives straightworward nice looking API to the end users.

Reply:
It **can** be implemented for general use-case of non-standard intallation place. I suggest first go with this solution that works in common use-cases.

Reply:
Or, this non-general solution can be made optional: so that the user have to enable it in MPP settings.

Reply:
Perhaps a better idea is to just handle this the same way as we're currently handling user macros. Just find, compile and load a less file from a predetermined location. Would that work for you?